edge-modules/TwinTester/Settings.cs

```diff
@@ -20,26 +20,32 @@ public class Settings
                     .AddEnvironmentVariables()
                     .Build();
 
-                string moduleId = configuration.GetValue<string>("TargetModuleId");
-                moduleId = string.IsNullOrWhiteSpace(moduleId) ? configuration.GetValue<string>("IOTEDGE_MODULEID", string.Empty) : moduleId;
+                string moduleId = configuration.GetValue<string>("IOTEDGE_MODULEID", string.Empty);
+                string targetModuleId = configuration.GetValue<string>("TargetModuleId", moduleId);
+
+                string analyzerUrl = configuration.GetValue<string>("AnalyzerUrl", "http://analyzer:15000");
+                string reporterUrl = configuration.GetValue<string>("testResultCoordinatorUrl", analyzerUrl);
 
                 return new Settings(
                     configuration.GetValue<string>("IOTEDGE_DEVICEID", string.Empty),
                     moduleId,
+                    targetModuleId,
                     configuration.GetValue<int>("TwinUpdateSize", 1),
                     configuration.GetValue<TimeSpan>("TwinUpdateFrequency", TimeSpan.FromSeconds(10)),
                     configuration.GetValue<TimeSpan>("TwinUpdateFailureThreshold", TimeSpan.FromMinutes(1)),
                     configuration.GetValue<TransportType>("TransportType", TransportType.Amqp_Tcp_Only),
-                    configuration.GetValue<Uri>("AnalyzerUrl", new Uri("http://analyzer:15000")),
+                    new Uri(reporterUrl),
                     configuration.GetValue<string>("ServiceClientConnectionString"),
                     configuration.GetValue<string>("StoragePath"),
                     configuration.GetValue<bool>("StorageOptimizeForPerformance", true),
-                    configuration.GetValue<TestMode>("TestMode", TestMode.TwinAllOperations));
+                    configuration.GetValue<TestMode>("TestMode", TestMode.TwinAllOperations),
+                    Option.Maybe(configuration.GetValue<string>("trackingId", null)));
```

Comment:
This is Optional because it is only used in Connectivity Tests - is that right?

Reply:
yes